libs/langchain/langchain/document_loaders/mongodb.py

```diff
@@ -0,0 +1,76 @@
+import asyncio
+import logging
+from typing import Dict, List, Optional
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+logger = logging.getLogger(__name__)
+
+
+class MongodbLoader(BaseLoader):
+    """Load MongoDB documents."""
+
+    def __init__(
+        self,
+        connection_string: str,
+        db_name: str,
+        collection_name: str,
+        *,
+        filter_criteria: Optional[Dict] = None,
+    ) -> None:
+        try:
+            from motor.motor_asyncio import AsyncIOMotorClient
+        except ImportError as e:
+            raise ImportError(
+                "Cannot import from motor, please install with `pip install motor`."
+            ) from e
+        if not connection_string:
+            raise ValueError("connection_string must be provided.")
+
+        if not db_name:
+            raise ValueError("db_name must be provided.")
+
+        if not collection_name:
+            raise ValueError("collection_name must be provided.")
+
+        self.client = AsyncIOMotorClient(connection_string)
+        self.db_name = db_name
+        self.collection_name = collection_name
+        self.filter_criteria = filter_criteria or {}
+
+        self.db = self.client.get_database(db_name)
+        self.collection = self.db.get_collection(collection_name)
+
+    def load(self) -> List[Document]:
+        """Load data into Document objects.
+
+        Attention:
```

Comment:
@saginawj i updated the description here. If you're interested, feel free to follow up with a PR to fix this implementation.

Steps:

1) Detect if there's a running loop
```
In [1]: import asyncio
In [2]: asyncio.get_running_loop()
```
2) 
-- If it doesn't exist then use asyncio.run with self.aload()
-- if it exists, strart a threadpool with 1 thread and have the thread run asyncio.run with self.aload()